### PR TITLE
purge all old archives regardless of done value

### PR DESCRIPTION
**Pull request by diosmosis**
we only care about the newest usable one

### CODE REVIEW

File: core/DataAccess/Model.php [hunk 1]
```diff
@@ -46,25 +46,20 @@ public function __construct(LoggerInterface $logger = null)
...
         $sql = "SELECT idsite, date1, date2, period, name,
                        GROUP_CONCAT(idarchive, '.', value ORDER BY ts_archived DESC) as archives
                   FROM `$archiveTable`
                  WHERE name LIKE 'done%'
-                   AND value IN (" . ArchiveWriter::DONE_INVALIDATED . ','
-                                   . ArchiveWriter::DONE_OK . ','
-                                   . ArchiveWriter::DONE_OK_TEMPORARY . ")
-                   AND idsite IN (" . implode(',', $idSites) . ")
-                 GROUP BY idsite, date1, date2, period, name";
+                   AND `value` NOT IN (" . ArchiveWriter::DONE_ERROR . ")
```
Comment (tsteur) [inline core/DataAccess/Model.php:61]: btw @diosmosis out of curiosity, do you maybe know why we don't delete an ERROR archive?

Comment (diosmosis) [reply core/DataAccess/Model.php:61]: I think it might be deleted somewhere else... I guess we can get rid of everything?

Comment (tsteur) [reply core/DataAccess/Model.php:61]: Merged it for now @diosmosis . @mattab do you maybe remember? I reckon it's otherwise not too important

Comment (diosmosis) [reply core/DataAccess/Model.php:61]: Oh, I know what it's for, this is the SELECT, so we're looking for archives that are usable and deleting everything older. If we SELECT for DONE_ERROR archives, then the latest might be DONE_ERROR and we end up deleting the usable archives. We can include DONE_ERROR here but we'd have to a little extra processing below.

I think they are currently deleted in another task via `Model::getTemporaryArchivesOlderThan()`.

Comment (tsteur) [reply core/DataAccess/Model.php:61]: makes sense, thanks 👍

**Comment by tsteur** [comment]
@diosmosis I just ran the task for us and I think it didn't clean up anything maybe because there are no reports marked as to be invalidated

image

and it seems it would only remove the ones that were invalidated 

image

Not sure this is maybe still an issue? But I suppose then the only solution would be to iterate over all archive tables for this which we maybe don't want as it could be quite resource intensive?

Not sure we're doing it already. Maybe it would help to invalidate all previous archives for the same site, date & period when we write a new OK archive?

**Comment by diosmosis** [comment]
@tsteur The fix is to run the command manually, otherwise we'd have to somehow add one date per existing archive table to ArchivesToPurgeDistributedList. Could do that in an update?

**Comment by diosmosis** [comment]
> Maybe it would help to invalidate all previous archives for the same site, date & period when we write a new OK archive?

I guess in the purge task, we can always do today/yesterday, that would solve the biggest source of dupes.

**Comment by tsteur** [comment]
That sounds good to do it in an update 👍  If I see this right in our DB it is only march and april that seems impacted by this so would only need to add two dates for these months maybe?

Currently executing this command to potentially fix it for now for us:

```
console core:purge-old-archive-data "2020-04-15" "2020-03-15" --exclude-ranges --skip-optimize-tables --include-year-archives  '
```

**Comment by tsteur** [comment]
> I guess in the purge task, we can always do today/yesterday, that would solve the biggest source of dupes.

👍  that should work and be quite simple. I guess today would be enough. Yesterday could be done maybe if it belongs to a different month? As `purgeInvalidatedArchivesFrom` looks at the entire month anyway

**Comment by diosmosis** [comment]
> I guess today would be enough. Yesterday could be done maybe if it belongs to a different month?

:+1:

**Pull request merged**